src/pdbvangogh/models.py

```diff
@@ -0,0 +1,136 @@
+        self.style_layers = style_layers
+        self.content_layers = content_layers
+        self.num_style_layers = len(style_layers)
+        self.vgg.trainable = False
+
+    def call(self, inputs):
+        "Expects float input in [0,1]"
+        inputs = inputs * 255.0
+        preprocessed_input = tf.keras.applications.vgg19.preprocess_input(inputs)
+        outputs = self.vgg(preprocessed_input)
+        style_outputs, content_outputs = (outputs[: self.num_style_layers], outputs[self.num_style_layers :])
+
+        style_outputs = [gram_matrix(style_output) for style_output in style_outputs]
+
+        content_dict = {content_name: value for content_name, value in zip(self.content_layers, content_outputs)}
+
+        style_dict = {style_name: value for style_name, value in zip(self.style_layers, style_outputs)}
+
+        return {"content": content_dict, "style": style_dict}
+
+
+def style_content_loss(outputs, style_targets, content_targets, num_content_layers, num_style_layers, style_weight, content_weight):
+    style_outputs = outputs["style"]
+    content_outputs = outputs["content"]
+    style_loss = tf.add_n([tf.reduce_mean((style_outputs[name] - style_targets[name]) ** 2) for name in style_outputs.keys()])
+    style_loss *= style_weight / num_style_layers
+
+    content_loss = tf.add_n([tf.reduce_mean((content_outputs[name] - content_targets[name]) ** 2) for name in content_outputs.keys()])
+    content_loss *= content_weight / num_content_layers
+    loss = style_loss + content_loss
+    return loss
+
+
+def am_i_in_a_jupyter_notebook():
+    try:
+        from IPython import get_ipython
+
+        # Check if the get_ipython function exists
+        # If we're in Jupyter, get_ipython will not be None
+        if "IPKernelApp" not in get_ipython().config:  # pragma: no cover
+            return False
+    except ImportError:
+        # IPython is not installed, definitely not in a notebook
+        return False
+    except AttributeError:
+        # IPython is installed but we're not in a Jupyter environment
+        return False
+    return True
+
+
+def style_transfer_vgg19(content_image, style_image, hyperparameters=data_models.vgg19_transfer_parameters()):
+    x = tf.keras.applications.vgg19.preprocess_input(content_image * 255)
+    x = tf.image.resize(x, (224, 224))
+    vgg = tf.keras.applications.VGG19(include_top=False, weights="imagenet")
+    content_layers = ["block5_conv2"]
+    style_layers = ["block1_conv1", "block2_conv1", "block3_conv1", "block4_conv1", "block5_conv1"]
+    num_content_layers = len(content_layers)
+    num_style_layers = len(style_layers)
+    style_extractor = vgg_layers(style_layers)
+    style_outputs = style_extractor(style_image * 255)
+    extractor = StyleContentModel(style_layers, content_layers)
+    results = extractor(tf.constant(content_image))
+    style_targets = extractor(style_image)["style"]
+    content_targets = extractor(content_image)["content"]
+    start = time.time()
```

Comment:
you're really only doing this once so not a big thing, but consider using a timing decorator or context manager if you start doing more timing / time measuring/reporting operations.  Also the advantage of the decoupling the implementation of the timing and display from the function itself